torch/csrc/jit/function.cpp

```diff
@@ -46,19 +45,11 @@ IValue Function::operator()(
 }
 
 void Function::ensure_defined() {
```

Comment:
There are call sites outside of the two callsites you've added error checking, is it valid to remove the checks there ? In `interpreter.cpp`, `schema_matching.cpp`, and `compiler.cpp`.

Reply:
Yes, the only way we can call things recursively is either by calling a method or by calling a function. To be clear: a `RecursiveMethodCallError` exception will still be thrown if we managed to create recursion some other way, it'll just won't be printed with a nice hint and source location.

Reply:
true lol